cmd/legacy/modern.go

```diff
@@ -207,7 +200,17 @@ func (l ModernFileSet) GetBlock(ts int64) (*hashmap.Map, error) {
 			V.PacketsSent = binary.BigEndian.Uint64(pktsSentBlock[i*8 : i*8+8])
 		}
 
-		data.SetOrUpdate(K, V.BytesRcvd, V.BytesSent, V.PacketsRcvd, V.PacketsSent)
+		if sip.Is4() && dip.Is4() {
```

Comment:
Fair point, with a few caveats:

- The hashmap doesn't know about `types.Key` (behind the scenes that's all just `[]byte` of course, but the hashmap is key agnostic (as long as it's a byte slice). The IPv4 / IPv6 split only happens on (sub)map level. I'm not 100% satisfied with the concept yet (because it's somewhere between generic and specific), but to keep options for later cleanup open I'd like to avoid making it _more_ specific.
- The if / else split still needs to happen because of the `.As4()` and `.As16()` calls for `netip.Addr`.

That being said, to make it more readable I've subsumed the logic under a private function and extended the aggregate hashmap by a new method that does not require internal knowledge about `types.Key` and still improves readability:
```
data.SetOrUpdate(
	newKeyFromNetIPAddr(sip, dip, dportBlock[i*2:i*2+2], protoBlock[i], isIPv4),
	isIPv4, V.BytesRcvd, V.BytesSent, V.PacketsRcvd, V.PacketsSent)
```